python/samples/getting_started_with_processes/step01_processes.py

```diff
@@ -0,0 +1,210 @@
+from semantic_kernel.processes.process_builder import ProcessBuilder
+from semantic_kernel.processes.process_function_target_builder import ProcessFunctionTargetBuilder
+
+
+class CommonEvents:
+    UserInputReceived = "UserInputReceived"
+    AssistantResponseGenerated = "AssistantResponseGenerated"
+
+
+class ChatBotEvents:
+    StartProcess = "startProcess"
+    IntroComplete = "introComplete"
+    AssistantResponseGenerated = "assistantResponseGenerated"
+    Exit = "exit"
+
+
+class UserInputState(KernelBaseModel):
+    user_inputs: list[str] = []
+    current_input_index: int = 0
+
+
+# Create a type variable for the generic state
+TState = TypeVar("TState", bound=KernelBaseModel)
+
+
+class KernelProcessStep(Generic[TState]):
+    def __init__(self):
+        self._state: TState | None = None
+
+    async def activate(self, state: "KernelProcessStepState[TState]"):
+        """Activates the step and sets the state."""
+        pass
+
+    def create_default_state(self) -> TState:
+        """Creates the default state object."""
+        return None
+
+    async def on_activate(self):
+        """To be overridden by subclasses if needed."""
+        pass
+
+
+class ScriptedUserInputStep(KernelProcessStep[UserInputState]):
+    class Functions:
+        GetUserInput = "get_user_input"
+
+    def create_default_state(self) -> "UserInputState":
+        """Creates the default UserInputState."""
+        return UserInputState()
+
+    def populate_user_inputs(self):
+        """Method to be overridden by the user to populate with custom user messages."""
+        pass
+
+    async def on_activate(self):
+        """This is called during the activation of the process step."""
+        self.populate_user_inputs()
+
+    async def activate(self, state: "KernelProcessStepState[TState]"):
+        """Activates the step and sets the state."""
+        state.state = state.state or self.create_default_state()
+        self._state = state.state
+        self.populate_user_inputs()
+        pass
+
+    @kernel_function(name=Functions.GetUserInput)
+    async def get_user_input(self, context: KernelProcessStepContext):
+        """Gets the user input."""
+        if not self._state:
+            raise ValueError("State has not been initialized")
+
+        user_message = self._state.user_inputs[self._state.current_input_index]
+
+        print(f"USER: {user_message}")
+
+        if "exit" in user_message:
+            await context.emit_event(KernelProcessEvent(id=ChatBotEvents.Exit, data=None))
+            return
+
+        self._state.current_input_index += 1
+
+        # Emit the user input event
+        await context.emit_event({"id": CommonEvents.UserInputReceived, "data": user_message})
+
+
+class ChatUserInputStep(ScriptedUserInputStep):
+    def populate_user_inputs(self):
+        """Override the method to populate user inputs for the chat step."""
+        if self._state is not None:
```

Comment:
This will get cleaned up, too -- renamed to `self.state`, but wasn't updated here.